cpp/include/cudf/detail/concatenate.cuh

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020-2021, NVIDIA CORPORATION.
+ * Copyright (c) 2020-2022, NVIDIA CORPORATION.
```

Comment:
The purpose of the changes in this file and in `cpp/src/copying/concatenate.cu` is to move the implementation of the `concatenate_masks` function into a detail method. I also fixed the Doxygen for a couple of detail methods that were incorrectly using `copydoc` but didn't match the source's signature at all.